docs/curate-text/process-data/language-management/index.md

```diff
@@ -12,57 +12,211 @@ modality: "text-only"
 
-- **Language detection** using FastText (176 languages) and CLD2 (used in HTML extraction pipelines)
-- **Stop word management** with built-in lists and customizable thresholds
-- **Special handling** for non-spaced languages (Chinese, Japanese, Thai, Korean)
-- **Language-specific** text processing and quality filtering
+## Complete Language Management Example
+
+Here's a comprehensive pipeline demonstrating language detection and filtering:
+
+```python
+from nemo_curator.pipeline import Pipeline
+from nemo_curator.stages.text.io.reader import JsonlReader
+from nemo_curator.stages.text.io.writer import JsonlWriter
+from nemo_curator.stages.text.modules import ScoreFilter
+from nemo_curator.stages.text.filters import FastTextLangId
+from nemo_curator.stages.function_decorators import processing_stage
+from nemo_curator.tasks import DocumentBatch
+
+# Create comprehensive language management pipeline
+pipeline = Pipeline(name="language_management_complete")
+
+# 1. Load input data
+pipeline.add_stage(
+    JsonlReader(file_paths="raw_data/", files_per_partition=4)
+)
+
+# 2. Detect languages with FastText
+pipeline.add_stage(
+    ScoreFilter(
+        FastTextLangId(
+            model_path="/models/lid.176.bin",
+            min_langid_score=0.6  # Medium-high confidence
+        ),
+        score_field="language"
+    )
+)
+
+# 3. Filter to English documents only
+@processing_stage(name="keep_english")
+@processing_stage(name="keep_english")
+def filter_english(batch: DocumentBatch) -> DocumentBatch:
+    import ast
+    df = batch.data
+    parsed = df["language"].apply(lambda v: ast.literal_eval(v) if isinstance(v, str) else v)
+    df["lang_code"] = parsed.apply(lambda p: str(p[1]))
+    df = df[df['lang_code'] == 'EN']
+    return DocumentBatch(data=df, task_id=batch.task_id, dataset_name=batch.dataset_name)
+
+pipeline.add_stage(filter_english)
+
+# 4. Export filtered, high-quality English documents
+pipeline.add_stage(JsonlWriter(path="curated_english/"))
+
+# Execute pipeline (uses XennaExecutor by default)
+results = pipeline.run()
+print("Language management pipeline completed!")
+```
+
+**Expected workflow:**
+1. Load multilingual JSONL documents
+2. Detect language with 60% minimum confidence
+3. Keep only English documents
+4. Export high-quality English dataset
+
+## Troubleshooting
+
+### Common Issues
+
+**FastText model not found:**
+```
+FileNotFoundError: [Errno 2] No such file or directory: '/path/to/lid.176.bin'
+```
+**Solution:** Download the model from [FastText Language Identification](https://fasttext.cc/docs/en/language-identification.html) and provide the correct absolute path.
+
+**Low detection accuracy:**
+```
+Many documents classified incorrectly
+```
+**Solution:** 
+- Increase `min_langid_score` to filter low-confidence predictions
+- Ensure input text is clean (remove HTML tags, special characters)
+- Check for very short documents (<50 chars) which are harder to classify
 
 ## Available Tools
```

Comment:
Since this page is about multilingual support with Curator, maybe we can mention the `MultilingualDomainClassifier` too?